application-templates/typescript/job/src/middleware/error.middleware.ts

```diff
@@ -1,31 +1,25 @@
 import { ErrorRequestHandler, NextFunction, Request, Response } from 'express';
 import CustomError from '../errors/custom.error';
 
-/**
- * Middleware for error handling
- * @param error The error object
- * @param req The express request
- * @param res The Express response
- * @param next
- * @returns
- */
-export const errorMiddleware = (
-  error: ErrorRequestHandler,
+
+export const errorMiddleware: ErrorRequestHandler = (
+  error: Error,
   req: Request,
   res: Response,
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   next: NextFunction
 ) => {
   if (error instanceof CustomError) {
-    if (typeof error.statusCode === 'number') {
-      res.status(error.statusCode).json({
+    
+      res.status(error.statusCode as number).json({
         message: error.message,
         errors: error.errors,
+        stack: error.stack,
```

Comment:
I think we should not expose the stack trace

Reply:
@abreu-ct I have seen such an example code

```js
res.json({
    message: err.message,
    // Only show stack trace in development mode
    stack: process.env.NODE_ENV === 'development' ? err.stack : undefined,
  });
  ```
  
So can we have a similar functionality: only if the `NODE_ENV` variable is set to `development` we return the stack trace? Are there any vulnerabilities linked to this kind of solution?

Reply:
Sorry @abreu-ct and @sfernengel. Maybe I didn't express myself correctly. 

In normal conditions, I would also not expose the stack trace, for obvious security reasons. But, in this case, as this is a template and will allow somehow the devs to do some code debugging, I thought this was not a problem. 

Something like this will also be caught in our certification process.

A solution like @sfernengel proposed could work in the short term.

Reply:
@sfernengel I would also go with your solution so that we don't expose stack trace in production

Reply:
I'm going to fix these changes and ask for a review again. Thanks!